amlb/datasets/file.py

```diff
@@ -138,37 +135,45 @@ def _extract_train_test_paths(self, dataset, fold=None, name=None):
     def __repr__(self):
         return repr_def(self)
 
-
     def extend_dataset_with_timeseries_config(self, dataset, dataset_config):
         dataset = deepcopy(dataset)
         dataset_config = deepcopy(dataset_config)
+        if dataset_config['forecast_horizon_in_steps'] is None:
+            raise AssertionError("Task definition for timeseries must include `forecast_horizon_in_steps`")
+        if dataset_config['seasonality'] is None:
+            raise AssertionError("Task definition for timeseries must include `seasonality`")
         if dataset_config['id_column'] is None:
-            log.warning("Warning: For timeseries task setting undefined `id_column` to `item_id`.")
-            dataset_config['id_column'] = "item_id"
-        if dataset_config['forecast_range_in_steps'] is None:
-            log.warning("Warning: For timeseries task setting undefined `forecast_range_in_steps` to `1`.")
-            dataset_config['forecast_range_in_steps'] = "1"
-
-        dataset.timestamp_column=dataset_config['timestamp_column']
-        dataset.id_column=dataset_config['id_column']
-        dataset.forecast_range_in_steps=int(dataset_config['forecast_range_in_steps'])
-
-        train_seqs_lengths = dataset.train.X.groupby(dataset.id_column).count()
-        test_seqs_lengths = dataset.test.X.groupby(dataset.id_column).count()
-        forecast_range_in_steps_mean_diff_train_test = int((test_seqs_lengths - train_seqs_lengths).mean())
-        forecast_range_in_steps_max_min_train_test = int(min(int(test_seqs_lengths.min()), int(train_seqs_lengths.min()))) - 1
-        if not dataset.forecast_range_in_steps == forecast_range_in_steps_mean_diff_train_test:
-            msg = f"Warning: Forecast range {dataset.forecast_range_in_steps}, does not equal mean difference between test and train sequence lengths {forecast_range_in_steps_mean_diff_train_test}."
-            log.warning(msg)
-        if not (test_seqs_lengths - train_seqs_lengths).var().item() == 0.:
-            msg = f"Error: Not all sequences of train and test set have same sequence length difference."
-            raise ValueError(msg)
-        if dataset.forecast_range_in_steps > forecast_range_in_steps_mean_diff_train_test:
-            msg = f"Error: Forecast range {dataset.forecast_range_in_steps} longer than difference between test and train sequence lengths {forecast_range_in_steps_mean_diff_train_test}."
-            raise ValueError(msg)
-        if dataset.forecast_range_in_steps > forecast_range_in_steps_max_min_train_test:
-            msg = f"Error: Forecast range {dataset.forecast_range_in_steps} longer than minimum sequence length + 1, {forecast_range_in_steps_max_min_train_test}."
-            raise ValueError(msg)
+            log.warning("Warning: For timeseries task, setting undefined `id_column` to `item_id`.")
+            dataset_config['id_column'] = 'item_id'
```

Comment:
Every dataset should contain `timestamp_column` and `id_column` - these are necessary to understand which rows in the CSV file correspond to each individual time series & sorting the values chronologically. I would expect that for most datasets these columns have the same default names (`timestamp` and `item_id`, respectively), so we don't need to copy the same code across all task definitions.

How about we change this code as follows?
```python
if dataset_config['id_column'] is None:
    if 'item_id' not in dataset.train.data:
        raise AssertionError('Task definition for timeseries must include `id_column` since the default ID column "item_id" is missing from the data.')
    else:
        log.warning("Warning: For timeseries task, setting undefined `id_column` to `item_id`.")
        dataset_config['id_column'] = 'item_id'
```

Alternatively, we could provide the default names for these columns somewhere else in the code.

Reply:
Sorry for being unclear - it seems reasonable enough to expect some kind of identifier column and some timestamp column, just maybe not that the `id_column` is called `item_id`. But as long as an exception is raised of `"item_id"` is not present, I don't see an objection to it being the default. In any case you can drop the inner 'else` and I would also add an outer else to check if the provided value (if there is one) is actually present in the data:

```python
if dataset_config['id_column'] is None:
    if 'item_id' not in dataset.train.data:
        raise ValueError('Task definition for timeseries must include `id_column` since the default ID column "item_id" is missing from the data.')
    log.warning("Warning: For timeseries task, setting undefined `id_column` to `item_id`.")
    dataset_config['id_column'] = 'item_id'
elif dataset_config['id_column'] not in dataset.train.data:
    raise ValueError(f"`id_column` set to '{dataset_config["id_column"]}' but is missing from the data.")
```

I considered
```python
dataset_config['id_column']  =  dataset_config['id_column']  or "item_id"
if dataset_config['id_column'] not in dataset.train.data:
    raise ValueError(f"`id_column` set to '{dataset_config["id_column"]}' but is missing from the data.")
```
but the extra warning message on setting a default and more precise error messages seem useful enough for the code to be more verbose.